### PR TITLE
stdin for multiprocessing

### PR DESCRIPTION
**Pull request by samuelcolvin**
I hoped #118 this was going to be fixed by #144 but it wasn't, after a bit of investigation I found the explanation [here](https://stackoverflow.com/questions/9178751/use-pdb-set-trace-in-a-script-that-reads-stdin-via-a-pipe#comment62352104_34687825).

@amirouche I'd be very grateful if you'd try and and check it works.

If I don't hear anything I'll merge in a couple of days.

### CODE REVIEW

File: aiohttp_devtools/runserver/watch.py [hunk 1]
```diff
@@ -108,7 +109,13 @@ def _start_process(self):
         act = 'Start' if self._change_count == 0 else 'Restart'
         logger.info('%sing dev server at http://%s:%s ●', act, self._config.host, self._config.main_port)
 
-        self._process = Process(target=serve_main_app, args=(self._config,))
+        try:
+            tty_path = os.ttyname(sys.stdin.fileno())
```
Comment (samuelcolvin) [inline aiohttp_devtools/runserver/watch.py]: I have no idea if this is a reliable way of getting the stdin tty path. It works for me on ubuntu.

File: aiohttp_devtools/runserver/serve.py [hunk 2]
```diff
@@ -90,42 +91,44 @@ def get_host(request):
...
+        modify_main_app(app, config)
+
+        loop.run_until_complete(check_port_open(config.main_port, loop))
+        handler = app.make_handler(
+            logger=dft_logger,
+            access_log_format='%r %s %b',
+            loop=loop,
+        )
+        co = asyncio.gather(
+            loop.create_server(handler, HOST, config.main_port, backlog=128),
```
Comment (asvetlov) [inline aiohttp_devtools/runserver/serve.py]: Againg `startup()` first than `create_server()`

File: aiohttp_devtools/runserver/serve.py [hunk 3]
```diff
@@ -90,42 +91,44 @@ def get_host(request):
...
+            loop.run_forever()
+        except KeyboardInterrupt:  # pragma: no cover
             pass
-        loop.run_until_complete(app.cleanup())
-    loop.close()
+        finally:
+            server.close()
+            loop.run_until_complete(server.wait_closed())
+            loop.run_until_complete(app.shutdown())
+            try:
```
Comment (asvetlov) [inline aiohttp_devtools/runserver/serve.py]: Tip: `with contextlib.suppress(asyncio.TimeoutError):` drops two non-covered lines :)

File: aiohttp_devtools/runserver/watch.py (hunk 1 above)
Comment (asvetlov) [reply aiohttp_devtools/runserver/watch.py]: `os.ttyname` is not available on Windows but works on any Unix system IMHO.

**Comment by samuelcolvin** [comment]
I don't have a windows machine, what's the alternative to `os.ttyname` on windows?

**Comment by asvetlov** [comment]
Windows has no tty concept at all.
The only alternative is using `subprocess` module instead of `multiprocessing` and inheriting all stdin, stdout and stderr IMHO.
Maybe I'm wrong, used to write code for Windows about decade ago.

**Pull request merged**